src/cmd/support_archive/troubleshoot.go

```diff
@@ -0,0 +1,52 @@
+package support_archive
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/src/cmd/troubleshoot"
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const troubleshootCollectorName = "troubleshoot"
+const TroublshootOutputFileName = "troubleshoot.txt"
+
+type troubleshootCollector struct {
+	collectorCommon
+
+	context    context.Context
+	apiReader  client.Reader
+	kubeConfig rest.Config
+	namespace  string
+}
+
+func newTroubleshootCollector(context context.Context, log logr.Logger, supportArchive tarball, namespace string, apiReader client.Reader, kubeConfig rest.Config) collector { //nolint:revive // argument-limit doesn't apply to constructors
+	return troubleshootCollector{
+		collectorCommon: collectorCommon{
+			log:            log,
+			supportArchive: supportArchive,
+		},
+		context:    context,
+		apiReader:  apiReader,
+		kubeConfig: kubeConfig,
+		namespace:  namespace,
+	}
+}
+
+func (t troubleshootCollector) Name() string {
+	return troubleshootCollectorName
+}
+
+func (t troubleshootCollector) Do() error {
```

Comment:
If this function never returns an error, why do we need it then in the signature?